pkg/controllers/provisioning/v1alpha1/reallocation/annotate.go

```diff
@@ -0,0 +1,97 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reallocation
+
+import (
+	"context"
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	utilsnode "github.com/awslabs/karpenter/pkg/utils/node"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type Annotator struct {
+	kubeClient client.Client
+}
+
+// Takes in a list of underutilized nodes and populates the nodes with the ProvisionerUnderutilized Label
+func (a *Annotator) LabelUnderutilized(ctx context.Context, nodes []*v1.Node) {
+	for _, node := range nodes {
+		persisted := node.DeepCopy()
+		node.Labels[v1alpha1.ProvisionerUnderutilizedKey] = "true"
+		if err := a.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
+			zap.S().Debugf("Continuing after failing to patch node %s, %s", node.Name, err.Error())
```

Comment:
I'm nervous about too much "continue after" behavior. In the provisioning cases, it makes sense, since we've already done things like launched real infrastructure. 

I think we probably want to bubble up the error and try again later, rather than continuing the nail the API Server which will likely result in similar failures.